src/WP_CLI/SearchReplacer.php

```diff
@@ -107,7 +119,7 @@ private function run_recursively( $data, $serialised, $recursion_level = 0, $vis
 						)
 					);
 				} else {
-					foreach ( $data as $key => $value ) {
+					foreach ( (array) $data as $key => $value ) {
```

Comment:
This seems like it will have negative unintended consequences and should be removed.

Reply:
no worries, removed with the latest commit!

Reply:
the problem here is that `foreach`ing directly over an object is not always safe in PHP and these objects like above show why. Objects in PHP don't always allow iteration and functions like `is_iterable()` check for the instance of `\Traversable` on objects & aren't exactly reliable.

So instead of casting, I've found this to be the safest way to iterate through these objects:

```
try {
foreach ( $data as $key => $value ) {
	$data->$key = $this->run_recursively( $value, false, $recursion_level + 1, $visited_data );
}
} catch ( \Error $e ) {
	\WP_CLI::warning(
	sprintf(
	'Skipping an inconvertible serialized object: "%s", replacements might not be complete.',
	$data
	)
);

throw new Exception();

...
```

currently works in all versions of PHP I've tested, but I'll need to change my behat tests to account for PHP 7.2, 7.4 & 8.0 throwing this new warning